ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/rest/batches/RestQueryTranslationTestBatch0.java

```diff
@@ -0,0 +1,37 @@
+package io.confluent.ksql.test.rest.batches;
+
+import io.confluent.common.utils.IntegrationTest;
+import io.confluent.ksql.integration.IntegrationTestHarness;
+import io.confluent.ksql.rest.server.TestKsqlRestApp;
+import io.confluent.ksql.test.rest.RestQueryTranslationTest;
+import io.confluent.ksql.test.rest.RestTestCase;
+import io.confluent.ksql.test.util.ThreadTestUtil.ThreadSnapshot;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+@Category({IntegrationTest.class})
+public class RestQueryTranslationTestBatch0 extends RestQueryTranslationTest {
+
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+  private static final TestKsqlRestApp REST_APP = createTestApp(TEST_HARNESS);
+  @ClassRule
+  public static final RuleChain CHAIN = createRuleChain(TEST_HARNESS, REST_APP);
+  private static final AtomicReference<ThreadSnapshot> STARTING_THREADS = new AtomicReference<>();
+
+  public RestQueryTranslationTestBatch0(String name,
```

Comment:
When I've thought about parallelizing the (R)QTTs, this approach was one I considered.  I wonder if there's a way to make this DRY instead of WET.  

E.g., maybe we could parametrize over the number of forks and then have each fork call `data(forks, 0-[forks-1])`?  

Maybe we need something from JUnit 5 to parallelize over parameterized tests...  Alternatively, maybe the pom tricks you are pulling would let us do something parallel in JUnit 4?

Maybe we should use a WET approach and be done...

Reply:
If we go the WET route, could all these classes be inner classes in one file?  Maybe that'd "look better"?  🤷

Reply:
I actually tried a lot of variations of this since I'm not a fan of these files either.  One thing that semi-works is just having them as static inner classes of `RestQueryTranslationTest`, but it didn't seem to fully have the parallelism I would have expected -- it says it's running them all at once, but it actually starts them seemingly sequentially.

I also tried having them all run as in a single JVM with a special junit runner with a threadpool, which also worked quite well.  The main issue is that it complicated the main file to be a bit more complex (have to assign each test to a running task), and that the thread checking logic was broken since there's no easy way look at the whole system's threads without forking the JVM.  The benefit is that it was a bit easier to just have a for loop in the setup which did everything and you could just have a single variable with the amount of parallelism.

So, while this is boilerplplate-y (would be nice to use code generation!), it's pretty simple to understand.  I ultimately chose simplicity over more parameterized (and also a solution that was able to fork processes).  I'm hoping to do the same thing for QTT (consistency!), so it would be nice to hash out feedback here for them both.

Reply:
Reading your comment I'm fine with the way things are